client/src/main/java/org/eclipse/hono/client/impl/HonoConnectionImpl.java

```diff
@@ -769,9 +832,20 @@ public void closeAndFree(
 
                     } else if (HonoProtonHelper.isLinkEstablished(sender)) {
 
-                        log.debug("sender open [target: {}, sendQueueFull: {}]", targetAddress, sender.sendQueueFull());
-                        // wait on credits a little time, if not already given
-                        if (sender.getCredit() <= 0) {
+                        log.debug("sender open [target: {}, sendQueueFull: {}, remote max-message-size: {}]",
+                                targetAddress, sender.sendQueueFull(), sender.getRemoteMaxMessageSize());
+                        final long remoteMaxMessageSize = Optional.ofNullable(sender.getRemoteMaxMessageSize())
+                                .map(UnsignedLong::longValue)
+                                .orElse(0L);
+                        if (remoteMaxMessageSize < clientConfigProperties.getMinMaxMessageSize()) {
+                            // peer won't accept our (biggest) messages
+                            sender.close();
+                            log.debug("peer does not support minimum message size [required: {}, supported: {}",
+                                    clientConfigProperties.getMinMaxMessageSize(), remoteMaxMessageSize);
+                            senderPromise.tryFail(new ClientErrorException(HttpURLConnection.HTTP_PRECON_FAILED,
+                                    "peer does not meet sender's minimum max-message-size requirement"));
```

Comment:
Maybe include the `remoteMaxMessageSize` in the exception text so that it appears in the surrounding tracing span (if one exists) via the logged error message? (So that the application doesn't needed to be restarted if not run with DEBUG log level in order to find out about the actual advertised value.)